### PR TITLE
Snappy gRPC compression

### PR DESCRIPTION
**Pull request by tirsen**
Add support for gRPC compression using Snappy. We are bandwidth constrained on both our vtgate's and our vttablet's. Given neither of these use particularly large amounts of RAM or CPU we can decrease our hardware footprint dramatically by decreasing bandwidth usage and co-locating on the same physical hardware.

### CODE REVIEW

**Comment by tirsen** [comment]
This PR contains 3 commits that are only loosely related: Remove cgzip, upgrade gRPC to 1.10.1 and add support for snappy. Let me know if you prefer separate PRs for all of these.

File: go/vt/grpcclient/snappy.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	"flag"
+	"io"
+
+	"github.com/golang/snappy"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/encoding"
+)
+
+var (
+	compression = flag.String("grpc_compression", "", "how to comrpess grpc, default: nothing, supported: snappy")
```
Comment (dweitzman) [inline go/vt/grpcclient/snappy.go]: Nit: comrpess -> compress

**Comment by alainjobart** [comment]
Please fix the typo, then LGTM. I'm not sure which travis is failing, it doesn't 'govendor sync' before building?

**Comment by michael-berlin** [comment]
> Please fix the typo, then LGTM. I'm not sure which travis is failing, it doesn't 'govendor sync' before building?

Travis doesn't do `govendor sync`.

That's by design. Instead, it uses what ever is the current state in the "bootstrap" Docker image.

i.e. you must update the "bootstrap" Docker image before merging a change against the vendor.json file.

**Comment by michael-berlin** [comment]
Images were rebuilt to include the snappy package.

I pushed the typo fix to tirsen's branch. Once Travis passed, I'll merge this.

**Comment by tirsen** [comment]
Thanks @michael-berlin !

**Pull request merged**